ui/src/treeDataProvider/AzureParentNode.ts

```diff
@@ -116,28 +128,24 @@ export class AzureParentNode<T extends IAzureParentTreeItem = IAzureParentTreeIt
     }
 
     public async addNodeToCache(node: AzureNode): Promise<void> {
-        if (this._cachedChildren) {
-            // set index to the last element by default
-            let index: number = this._cachedChildren.length;
-            // tslint:disable-next-line:no-increment-decrement
-            for (let i: number = 0; i < this._cachedChildren.length; i++) {
-                if (node.treeItem.label.localeCompare(this._cachedChildren[i].treeItem.label) < 1) {
-                    index = i;
-                    break;
-                }
+        // set index to the last element by default
+        let index: number = this._cachedChildren.length;
+        // tslint:disable-next-line:no-increment-decrement
+        for (let i: number = 0; i < this._cachedChildren.length; i++) {
+            if (node.treeItem.label.localeCompare(this._cachedChildren[i].treeItem.label) < 1) {
+                index = i;
+                break;
             }
-            this._cachedChildren.splice(index, 0, node);
-            await this.treeDataProvider.refresh(this, false);
         }
+        this._cachedChildren.splice(index, 0, node);
+        await this.treeDataProvider.refresh(this, false);
     }
 
     public async removeNodeFromCache(node: AzureNode): Promise<void> {
-        if (this._cachedChildren) {
-            const index: number = this._cachedChildren.indexOf(node);
-            if (index !== -1) {
-                this._cachedChildren.splice(index, 1);
-                await this.treeDataProvider.refresh(this, false);
-            }
+        const index: number = this._cachedChildren.indexOf(node);
+        if (index !== -1) {
+            this._cachedChildren.splice(index, 1);
+            await this.treeDataProvider.refresh(this, false);
```

Comment:
Is this code mainly because the Set data structure is not sorted by default? (TIL it stores values in insertion order :P)

Reply:
Yes